src/providers/pdal/qgspdalprovider.cpp

```diff
@@ -0,0 +1,151 @@
+/***************************************************************************
+                         qgspdaldataprovider.cpp
+                         -----------------------
+  Date                 : November 2020
+  Copyright            : (C) 2020 by Peter Petrik
+  Email                : zilolv at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgis.h"
+#include "qgspdalprovider.h"
+#include "qgspdaldataitems.h"
+#include "qgsruntimeprofiler.h"
+#include "qgsapplication.h"
+
+#include <pdal/io/LasReader.hpp>
+#include <pdal/io/LasHeader.hpp>
+#include <pdal/Options.hpp>
+
+#define PROVIDER_KEY QStringLiteral( "pdal" )
+#define PROVIDER_DESCRIPTION QStringLiteral( "PDAL point cloud data provider" )
+
+QgsPdalProvider::QgsPdalProvider(
+  const QString &uri,
+  const QgsDataProvider::ProviderOptions &options,
+  QgsDataProvider::ReadFlags flags )
+  : QgsPointCloudDataProvider( uri, options, flags )
+{
+  std::unique_ptr< QgsScopedRuntimeProfile > profile;
+  if ( QgsApplication::profiler()->groupIsActive( QStringLiteral( "projectload" ) ) )
+    profile = qgis::make_unique< QgsScopedRuntimeProfile >( tr( "Open data source" ), QStringLiteral( "projectload" ) );
+
+  mIsValid = load( uri );
+}
+
+QgsPdalProvider::~QgsPdalProvider() = default;
+
+QgsCoordinateReferenceSystem QgsPdalProvider::crs() const
+{
+  return mCrs;
+}
+
+QgsRectangle QgsPdalProvider::extent() const
+{
+  return mExtent;
+}
+
+bool QgsPdalProvider::isValid() const
+{
+  return mIsValid;
+}
+
+QString QgsPdalProvider::name() const
+{
+  return QStringLiteral( "pdal" );
+}
+
+QString QgsPdalProvider::description() const
+{
+  return QStringLiteral( "Point Clouds PDAL" );
+}
+
+QgsPointCloudIndex *QgsPdalProvider::index() const
+{
+  // TODO automatically generate EPT index
+  return nullptr;
+}
+
+bool QgsPdalProvider::load( const QString &uri )
+{
+  pdal::Option las_opt( "filename", uri.toStdString() );
+  pdal::Options las_opts;
+  las_opts.add( las_opt );
+  pdal::PointTable table;
+  pdal::LasReader las_reader;
+  las_reader.setOptions( las_opts );
+  las_reader.prepare( table );
+  pdal::PointViewSet point_view_set = las_reader.execute( table );
```

Comment:
Oh I think we need to avoid this call - it will read all points from the file and store them in memory - this can take many seconds. It should be possible to get the list of dimensions without actually reading everything. E.g. Entwine displays the list of dimensions before starting indexing.

Reply:
removed, we do not need it here at all